src/realmModels/Observation.js

```diff
@@ -371,27 +378,45 @@ class Observation extends Realm.Object {
       fields: { id: true }
     };
 
-    const response = await createObservation( uploadParams, options );
+    let response;
+
+    const wasPreviouslySynced = Observation.wasPreviouslySyncedObservation( realm, obs );
+
+    if ( wasPreviouslySynced ) {
+      response = await updateObservation( {
+        id: newObs.uuid,
+        ignore_photos: true,
+        observation: { ...newObs },
+        fields: { id: true }
+      }, options );
+    } else {
+      response = await createObservation( uploadParams, options );
+    }
+
     await Observation.markRecordUploaded( obs.uuid, "Observation", response, realm );
     const { id } = response.results[0];
-    if ( obs?.observationPhotos?.length > 0 ) {
+
+    // skip updating evidence for already synced observations
+    // since these .update API calls aren't yet available in v2
+
+    if ( obs?.observationPhotos?.length > 0 && !wasPreviouslySynced ) {
       return Observation.uploadEvidence(
         obs.observationPhotos,
         "ObservationPhoto",
         ObservationPhoto.mapPhotoForUpload,
         id,
-        inatjs.observation_photos,
+        wasPreviouslySynced ? inatjs.observation_photos.update : inatjs.observation_photos.create,
```

Comment:
This use of `wasPreviouslySynced` is redundant, right? You know it's false from line 402.